Churn data analasis
1. importing the data to a variable df

In [27]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import f_oneway
from scipy.stats import chi2_contingency
df = pd.read_csv(r"extra\onlinefoods.csv")
df.head()

Age  Gender Marital Status Occupation  Monthly Income  \
0   20  Female         Single    Student       No Income   
1   24  Female         Single    Student  Below Rs.10000   
2   22    Male         Single    Student  Below Rs.10000   
3   22  Female         Single    Student       No Income   
4   22    Male         Single    Student  Below Rs.10000   

  Educational Qualifications  Family size  latitude  longitude  Pin code  \
0              Post Graduate            4   12.9766    77.5993    560001   
1                   Graduate            3   12.9770    77.5773    560009   
2              Post Graduate            3   12.9551    77.6593    560017   
3                   Graduate            6   12.9473    77.5616    560019   
4              Post Graduate            4   12.9850    77.5533    560010   

  Output   Feedback Unnamed: 12  
0    Yes   Positive         Yes  
1    Yes   Positive         Yes  
2    Yes  Negative          Yes  
3    Yes   Positive         Yes  
4    Yes   Positive         Yes

doing some primary analysis .

In [28]:
df.shape

(388, 13)

there are **388 rows and 21 columns** in the csv file.

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 388 entries, 0 to 387
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Age                         388 non-null    int64  
 1   Gender                      388 non-null    object 
 2   Marital Status              388 non-null    object 
 3   Occupation                  388 non-null    object 
 4   Monthly Income              388 non-null    object 
 5   Educational Qualifications  388 non-null    object 
 6   Family size                 388 non-null    int64  
 7   latitude                    388 non-null    float64
 8   longitude                   388 non-null    float64
 9   Pin code                    388 non-null    int64  
 10  Output                      388 non-null    object 
 11  Feedback                    388 non-null    object 
 12  Unnamed: 12                 388 non-null    object 
dtypes: float64(2), int64(3), object(8)


In [30]:
df.describe()

Age  Family size    latitude   longitude       Pin code
count  388.000000   388.000000  388.000000  388.000000     388.000000
mean    24.628866     3.280928   12.972058   77.600160  560040.113402
std      2.975593     1.351025    0.044489    0.051354      31.399609
min     18.000000     1.000000   12.865200   77.484200  560001.000000
25%     23.000000     2.000000   12.936900   77.565275  560010.750000
50%     24.000000     3.000000   12.977000   77.592100  560033.500000
75%     26.000000     4.000000   12.997025   77.630900  560068.000000
max     33.000000     6.000000   13.102000   77.758200  560109.000000

In [31]:
for i in df:
    if df[i].dtype == 'object':
        print(i,df[i].unique())

Gender ['Female' 'Male']
Marital Status ['Single' 'Married' 'Prefer not to say']
Occupation ['Student' 'Employee' 'Self Employeed' 'House wife']
Monthly Income ['No Income' 'Below Rs.10000' 'More than 50000' '10001 to 25000'
 '25001 to 50000']
Educational Qualifications ['Post Graduate' 'Graduate' 'Ph.D' 'Uneducated' 'School']
Output ['Yes' 'No']
Feedback ['Positive' 'Negative ']
Unnamed: 12 ['Yes' 'No']


    this shows all the unique values in each column.
    the total charges column is incorrectly stored as string value.

droping hte unamed column.

In [33]:
df.drop(columns='Unnamed: 12',inplace=True)

**EDA**
1) make a box plot for each numeric data ot check for outliers. (nothing intresting)
2) make bar chart for each catogarical data to check for abnormal distributions. (nothing intresting)

**Feature selection**

    intresting columns to consider based on EDA: 
    Contract
    online security and techsupport
    StreamingTv and steaming Movies
    MonthlyCharges
    tenure
    SeniorCitizen
    Dependents
    InternetService

In [ ]:
# df_new = df[['Contract','MonthlyCharges','tenure','Dependents','SeniorCitizen','InternetService','StreamingTV','StreamingMovies','TechSupport','OnlineSecurity','Churn']]

In [ ]:
# df_new.to_csv('extra.csv')